clients/typescript/src/client/validation/validation.ts

```diff
@@ -0,0 +1,20 @@
+import * as z from 'zod'
+
+export function validate<I>(i: I, schema: z.ZodTypeAny): I {
+  const parsedObject = schema.parse(i)
+
+  function deepOmit(obj: Record<string, any>) {
+    Object.keys(obj).map((key) => {
```

Comment:
```suggestion
    Object.keys(obj).foreach((key) => {
```

since we're not using `return` in the function